### PR TITLE
feat: update the Linux /window function app at v3 provider for doc and testcase

### PR DESCRIPTION
**Pull request by AlphaWong**
related to
#16592

### CODE REVIEW

**Comment by AlphaWong** [comment]
awaiting the CI to check the testcase passed or not.

**Comment by AlphaWong** [comment]
fix the code based on the error found by CI

**Comment by AlphaWong** [comment]
this function app naming issue still exists in the latest version v4 SDK

# reference
https://github.com/Azure/azure-functions-host/wiki/Host-IDs#host-id-collisions

**Comment by AlphaWong** [comment]
confirmed by m$ the sdk v4 still have this issue. Plz fix up the d95245fd2 commit with this one.

**Comment by AlphaWong** [comment]
m$ Offical document has been updated

https://github.com/MicrosoftDocs/azure-docs/commit/c00ef4f2fca91e78ee0a19c9bd2eabba2ee3189c

file: articles/azure-resource-manager/management/resource-name-rules.md

File: website/docs/r/function_app.html.markdown [hunk 1]
```diff
@@ -191,7 +191,7 @@ resource "azurerm_function_app" "example" {
 
 The following arguments are supported:
 
-* `name` - (Required) Specifies the name of the Function App. Changing this forces a new resource to be created. Limit the function name to 32 characters to avoid naming collisions. For more information about [Function App naming rule](https://docs.microsoft.com/azure/azure-resource-manager/management/resource-name-rules#microsoftweb)
+* `name` - (Required) Specifies the name of the Function App. Changing this forces a new resource to be created. Limit the function name to 32 characters to avoid naming collisions. For more information about [Function App naming rule](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftweb) and [Host ID Collisions](https://github.com/Azure/azure-functions-host/wiki/Host-IDs#host-id-collisions)
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:194]: This resource is deprecated, so shouldn't be updated here.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @AlphaWong 
Thanks for this PR, and apologies for the delay in review.

Unfortunately the changes proposed in this PR won’t work as expected as warnings are not (currently) surfaced by the provider in any way that would be useful to users of this resource. 

The documentation, however, should be updated to make users aware of the technical limitation in the Service Side. The full values are stored/returned, but only the first 32 chars are used to uniquely identify the name.  Typically this is done as a `NOTE` to draw attention to the extra information, I’ve left a suggestion below for such.  If you can revert the changes to the validation and update the docs as per the suggestion, I think this will be good to merge :+1:

Thanks again!

File: website/docs/r/function_app.html.markdown (hunk 1 above)
Comment (AlphaWong) [reply website/docs/r/function_app.html.markdown]: good point. let me only include the doc updates

Comment (AlphaWong) [reply website/docs/r/function_app.html.markdown]: I make a new commit. plz help me to squash the commits to one.

**Review by katbyte — APPROVED** [review]
LGTM 🌻

**Pull request merged**